crates/fluvio-spu/src/core/global_context.rs

```diff
@@ -478,5 +481,41 @@ mod file_replica {
                 error!("leader switch failed: {}", err);
             }
         }
+
+        /// Delete consumers offset for given replica if it is leader consumer
+        async fn delete_consumers_offset(&self, replica: &Replica) -> anyhow::Result<()> {
+            let consumers_replica_id =
+                ReplicaKey::new(CONSUMER_STORAGE_TOPIC, <PartitionId as Default>::default());
+            let Some(ref replica_consumer) = self.leaders_state().get(&consumers_replica_id).await
```

Comment:
let's move this as separate api something like `is_consumer_leader` or something like that

Reply:
moved to ReplicaLeadersState